crates/html_node/src/lib.rs

```diff
@@ -39,6 +39,22 @@ fn init() {
     }
 }
 
+#[napi_derive::napi(object)]
+#[derive(Debug, Serialize)]
+pub struct Diagnostic {
+    pub level: String,
+    pub message: String,
+    pub span: serde_json::Value,
+}
+
+#[napi_derive::napi(object)]
+#[derive(Debug, Serialize)]
+pub struct TransformOutput {
+    pub code: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub errors: Option<Vec<Diagnostic>>,
```

Comment:
Oh can you change this to `Vec<Diagnostic>`?

Reply:
@kdy1 Yep, but we use the same logic for `map`, so I think we need the same for errors too, otherwise it was a little strange

Reply:
At where?

Reply:
I guess `map` is `String` or `SourceMap`, and it's different from `Vec`.

Reply:
Hm, but why we don't return `undefined` for https://github.com/swc-project/bindings/blob/main/crates/css_node/src/lib.rs#L37 when it is `None`, so we will always have `code` and `map` options? And we can remove `Option` too here - i.e. we will always all fields (some of them will be undefined or empty).

Otherwise, it gets a little weird - at some points we return some fields, at others we omit them, it will look rather strange for developers.

just thoughts

Reply:
We are returning `undefined` implicitly

Reply:
JSON doesn't have `undefined`

Reply:
Yeah, I undestand this, I think we can solve it on JS side :smile: But yep, let's change it to `Vec<Diagnostic>`